### PR TITLE
2440: Fix infinite rerender cycle for links without inapp-browser

### PR DESCRIPTION
**Pull request by ztefanie**
### Short description

Added description on how to reproduce this bug to the issue.

### Proposed changes

There was endless rerendering for integreat.app links, if there is no inAppBrowser available, remove it by adjusting the checks.

### Resolved issues

Fixes: #2440 

---

### CODE REVIEW

**Review by LeandraH — APPROVED** [review]
Very cool, thanks for figuring out how to reproduce the original issue! 

Tested in emulated Android

File: native/src/utils/openExternalUrl.ts [hunk 1]
```diff
@@ -13,9 +13,15 @@ const openExternalUrl = async (rawUrl: string, showSnackbar: (snackbar: Snackbar
...
 
+  const noInAppBrowserAvailableAndIntegreatLink =
+    !(await InAppBrowser.isAvailable()) && encodedUrl.includes(buildConfig().hostName)
+  const canBeOpenedWithInAppBrowser = (await InAppBrowser.isAvailable()) && ['https:', 'http:'].includes(protocol)
+  const canBeOpenedWithOtherApp = await Linking.canOpenURL(encodedUrl)
+
   try {
-    // Custom tabs are not available in all browsers and support only http and https
-    if ((await InAppBrowser.isAvailable()) && ['https:', 'http:'].includes(protocol)) {
+    if (noInAppBrowserAvailableAndIntegreatLink) {
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply native/src/utils/openExternalUrl.ts:22]: I did this on purpose, it is a principle called "early exit", read more [here](https://www.linkedin.com/pulse/early-exit-java-enhancing-efficiency-readability-saeed-anabtawi/) if interessted

File: native/src/utils/openExternalUrl.ts [hunk 2]
```diff
@@ -13,9 +13,15 @@ const openExternalUrl = async (rawUrl: string, showSnackbar: (snackbar: Snackbar
   const encodedUrl = encodeURI(rawUrl)
   const { protocol } = new URL(encodedUrl)
 
+  const noInAppBrowserAvailableAndIntegreatLink =
+    !(await InAppBrowser.isAvailable()) && encodedUrl.includes(buildConfig().hostName)
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply native/src/utils/openExternalUrl.ts:17]: Nice good point, adjusted it.

File: native/src/utils/openExternalUrl.ts [hunk 3]
```diff
@@ -26,20 +32,16 @@ const openExternalUrl = async (rawUrl: string, showSnackbar: (snackbar: Snackbar
...
+        signal: {
+          name: OPEN_OS_LINK_SIGNAL_NAME,
+          url: encodedUrl,
+        },
+      })
+      await Linking.openURL(encodedUrl)
     } else {
-      const canOpen = await Linking.canOpenURL(encodedUrl)
-
-      if (canOpen) {
```
[the remark this answers is not in the document]
Comment (ztefanie) [reply native/src/utils/openExternalUrl.ts:32]: I restructered it, as I wanted to get rid of the nested if-else. I named the variables so it is clear, what is going on, added the last line of your comment, which hints to preventing an infinite loop with this check.

File: native/src/utils/openExternalUrl.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (LeandraH) [reply native/src/utils/openExternalUrl.ts:22]: Sure but we also have an exit at the end, so I'm not sure how much sense it makes to apply the principle half-way here. But anyway, legibility is often subjective, I'm okay with it the way it is.

**Review by steffenkleinle — APPROVED** [review]

**Pull request merged**